MusicTaste/MusicTaste/MatchingAlgorithm.m

```diff
@@ -0,0 +1,93 @@
+#import "MatchingAlgorithm.h"
+#import <Foundation/Foundation.h>
+#import <Parse/Parse.h>
+#import "AppDelegate.h"
+#import "dispatch/dispatch.h"
+
+@implementation MatchingAlgorithm
+
++ (instancetype)shared {
```

Comment:
Creates a shared instance of Algorithm file

Reply:
After further examination, I decide to still include the singleton; however, the alternative would be to have static functions and call those within other viewcontrollers since it would be public. However, I wanted the liberty to use variables within the file by calling on self and not have it static. This is why I kept the static variable and made the other functions non-static.

Reply:
> the alternative would be to have static functions and call those within other viewcontrollers since it would be public

Love this! Another alternative would be to pass the instance around everywhere. (Which can sometimes get messy if it's used in many different places in the app code.)

> variables within the file by calling on self and not have it static

Out of curiosity why did you want to avoid having static class variables?